src/components/SwitchButton/index.tsx

```diff
@@ -14,37 +14,43 @@ import Box, { BoxProps } from '../Box'
 import Tooltip from '../Tooltip'
 
 const variants = {
-  segment: ({ theme }: { theme: Theme }) => css`
-    font-size: 14;
-    height: 40px;
-    transition: none;
-    border-radius: 4px;
-    box-shadow: none;
-
-    &:hover,
-    &:focus {
+  segment: (theme: Theme) => css`
+    &[data-variant='segment'] {
+      font-size: 14;
+      height: 40px;
+      transition: none;
+      border-radius: 4px;
       box-shadow: none;
-      border-color: ${theme.colorsDeprecated.transparent};
-      color: ${theme.colorsDeprecated.gray700};
-    }
+      border-color: transparent;
+      background-color: ${theme.colors.neutral.backgroundWeak};
+      color: ${theme.colors.neutral.text};
 
-    &[aria-checked='true'] {
-      background-color: ${theme.colorsDeprecated.primary};
-      color: ${theme.colorsDeprecated.white};
-      :hover {
-        color: ${theme.colorsDeprecated.white};
+      &:hover,
+      &:focus {
+        border-color: transparent;
+        box-shadow: none;
       }
-    }
 
-    &[aria-checked='false'] {
-      background-color: white;
-      color: ${theme.colorsDeprecated.gray700};
-      border-color: ${theme.colorsDeprecated.transparent};
-      :hover,
-      :focus {
-        color: ${theme.colorsDeprecated.gray700};
-        border-color: ${theme.colorsDeprecated.transparent};
-        box-shadow: none;
+      &[aria-checked='true'] {
+        background-color: ${theme.colors.primary.backgroundStrong};
+        color: ${theme.colors.neutral.textStrong};
+      }
+
+      &[aria-disabled='true'] {
+        color: ${theme.colors.neutral.textDisabled};
+
+        &[aria-checked='true'] {
+          color: ${theme.colors.neutral.textDisabled};
+          background-color: ${theme.colors.neutral.backgroundDisabled};
+        }
+      }
+
+      &:not([aria-checked='true']):not([aria-disabled='true']) {
```

Comment:
Can't we do this? It's only not supported in IE https://developer.mozilla.org/fr/docs/Web/CSS/:not#compatibilit%C3%A9_des_navigateurs

```suggestion
      &:not([aria-checked='true'], [aria-disabled='true']) {
```